lib/common/web-view-events.ts

```diff
@@ -0,0 +1,36 @@
+export const webViewEvents: Record<string, string[]> = {
+  'load-commit': ['url', 'isMainFrame'],
+  'did-attach': [],
+  'did-finish-load': [],
+  'did-fail-load': ['errorCode', 'errorDescription', 'validatedURL', 'isMainFrame', 'frameProcessId', 'frameRoutingId'],
+  'did-frame-finish-load': ['isMainFrame', 'frameProcessId', 'frameRoutingId'],
+  'did-start-loading': [],
+  'did-stop-loading': [],
+  'dom-ready': [],
+  'console-message': ['level', 'message', 'line', 'sourceId'],
+  'context-menu': ['params'],
+  'devtools-opened': [],
+  'devtools-closed': [],
+  'devtools-focused': [],
+  'new-window': ['url', 'frameName', 'disposition', 'options'],
+  'will-navigate': ['url'],
+  'did-start-navigation': ['url', 'isInPlace', 'isMainFrame', 'frameProcessId', 'frameRoutingId'],
+  'did-navigate': ['url', 'httpResponseCode', 'httpStatusText'],
+  'did-frame-navigate': ['url', 'httpResponseCode', 'httpStatusText', 'isMainFrame', 'frameProcessId', 'frameRoutingId'],
+  'did-navigate-in-page': ['url', 'isMainFrame', 'frameProcessId', 'frameRoutingId'],
+  'focus-change': ['focus', 'guestInstanceId'],
+  close: [],
+  crashed: [],
+  'render-process-gone': ['details'],
```

Comment:
I noticed that `render-process-gone` is new here; it was not in `WEB_VIEW_EVENTS` before this commit moved it from `guest-view-internal.ts`. Is this a fix that needs to be backported?

The `render-process-gone` event itself was added in v10.0.0 (#23560) and was backported to v9.1.0 and v8.4.0.

I’m trying to debug a crash (zulip/zulip-desktop#1016) with a trace that shows [this `WEB_VIEW_EVENTS[eventKey].forEach`](https://github.com/electron/electron/blob/v9.3.0/lib/renderer/web-view/guest-view-internal.ts#L55) being undefined, and I wonder if this might be because `eventKey` is `"render-process-gone"`.